### PR TITLE
[feature] simpler cache size configuration

### LINKED ISSUE #2049: [feature] simpler cache configuration
currently our cache configuration is very complex. it requires twiddling the dials of maximum number of objects per object type, which unless you have some knowledge of our codebase must be quite intimidating ...

the proposed change is that instead we have a singular cache memory limit target variable, and we *aim* to keep our caches under this size using some estimates of average model size during runtime. we can have finer tuned dials that allow twiddling cache memory allocation of say accounts : statuses, but we should leave these out of the example config and have most people rely on the defaults of them. in time we can add docs for this. the cache memory limit target variable should be enough for most.

### PR DESCRIPTION
**Pull request by NyaaaWhatsUpDoc**
## Description

- updates caches to remove TTL checking + sweeping (still keeps to a maximum size)
- removes the many old, confusing cache configuration variables
- adds a singular `cache.memory-target` configuration variable to control cache sizes
- adds "hidden" mem ratio per cache configuration, not included in the example config

closes #2049 

## Checklist

Please put an x inside each checkbox to indicate that you've read and followed it: `[ ]` -> `[x]`

If this is a documentation change, only the first checkbox must be filled (you can delete the others if you want).

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [x] I/we have made any necessary changes to documentation.
- [ ] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

**Comment by NyaaaWhatsUpDoc** [comment]
Converting to draft as there's still some more specifics to figure out.

**Comment by NyaaaWhatsUpDoc** [comment]
basically, when caches get full then every single write operation also involves an evict which is more expensive. i'm experimenting adding a regular sweep (not TTL, in fact this simpler method has some advantages over TTL) that checks caches are within 80% of their capacity once a minute, else truncating them. and this would all happen from a singular background goroutine which is a bonus too.

File: example/config.yaml [hunk 1]
```diff
@@ -231,111 +231,13 @@ db-sqlite-cache-size: "8MiB"
...
-    webfinger-max-size: 250
-    webfinger-ttl: "24h"
-    webfinger-sweep-freq: "15m"
+  # cache.memory-target sets a target limit that
+  # the application will try to keep it's caches
+  # within. This is based on estimated sizes of
+  # in-memory objects, and so NOT AT ALL EXACT.
+  # Examples: ["100MiB", "200MiB", "500MiB", "1GiB"]
+  # Default: "200MiB"
+  memory-target: "200MiB"
```
Comment (tsmethurst) [inline example/config.yaml:240]: Damn so much cleaner!

File: go.mod [hunk 2]
```diff
@@ -16,6 +16,7 @@ require (
 	codeberg.org/gruf/go-runners v1.6.1
 	codeberg.org/gruf/go-sched v1.2.3
 	codeberg.org/gruf/go-store/v2 v2.2.2
+	github.com/DmitriyVTitov/size v1.5.0
```
Comment (tsmethurst) [inline go.mod:19]: Don't forget to add this new dependency to the readme with the others

File: example/config.yaml (hunk 1 above)
Comment (NyaaaWhatsUpDoc) [reply example/config.yaml:240]: right!!

File: internal/config/defaults.go [hunk 3]
```diff
@@ -126,111 +126,42 @@ var Defaults = Configuration{
...
+
+		// NOTE: these weighted by a totally
+		// assorted mixture of priority, and
+		// manual twiddling to get the generated
+		// cache capacity ratios within normal
+		// amounts dependent on relative sizes.
+		//
+		// when TODO items in the size.go source
+		// file have been addressed, these should
+		// be able to make some more sense :D
```
Comment (tsmethurst) [inline internal/config/defaults.go:139]: Could you add a little more explanation somewhere of what these numbers mean? How they relate to each other etc?

File: cmd/gotosocial/action/server/server.go [hunk 4]
```diff
@@ -117,6 +119,13 @@ var Start action.GTSAction = func(ctx context.Context) error {
 	state.Workers.Start()
 	defer state.Workers.Stop()
 
+	// Add a task to the scheduler to sweep caches.
+	// Frequency = 1 * minute
+	// Threshold = 80% capacity
+	sweep := func(time.Time) { state.Caches.Sweep(80) }
```
[the remark this answers is not in the document]
Comment (NyaaaWhatsUpDoc) [reply cmd/gotosocial/action/server/server.go:125]: Yep! it basically does this internally:
```golang
		// Calculate number of cache items to truncate.
		max := (perc / 100) * float64(c.Cache.Cap())
		diff := c.Cache.Len() - int(max)
		if diff <= 0 {
			return
		}

		// Truncate by calculated length.
		items = c.truncate(diff)
```

File: internal/cache/size.go [hunk 5]
```diff
@@ -0,0 +1,500 @@
...
+	"github.com/superseriousbusiness/gotosocial/internal/config"
+	"github.com/superseriousbusiness/gotosocial/internal/gtsmodel"
+	"github.com/superseriousbusiness/gotosocial/internal/id"
+)
+
+const (
+	// example data values.
+	exampleID   = id.Highest
+	exampleURI  = "https://social.bbc/users/ItsMePrinceCharlesInit"
+	exampleText = `
```
Comment (tsmethurst) [inline internal/cache/size.go:37]: Having a normal one were we?

Comment (NyaaaWhatsUpDoc) [reply internal/cache/size.go:37]: so normal you wouldn't believe it

File: cmd/gotosocial/action/server/server.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (tsmethurst) [reply cmd/gotosocial/action/server/server.go:125]: Nice!

**Pull request merged**